pkg/controller/gitopsconfig/controller_test.go

```diff
@@ -595,21 +606,28 @@ func TestCreateJob(t *testing.T) {
 	}
 
 	numberOfJobs := 2
-	var wg sync.WaitGroup
-	for i := 1; i <= numberOfJobs; i++ {
-		wg.Add(1)
-		// in order to replicate reconciler being called for creating an another job.
-		go func() {
-			r.Reconcile(req)
-			wg.Done()
-		}()
-		time.Sleep(3 * time.Second)
-	}
-	wg.Wait()
+	r.Reconcile(req)
+
+	// fakeclient is not updating the job status , inorder to test race condition between the jobs
+	//updating the job status manually
+	startTime := metav1.Now()
+	job := findRunningJob(cl)
+	job.Status.Active = 1
+	job.Status.Succeeded = 0
+	job.Status.Failed = 0
+	job.Status.StartTime = &startTime
 
-	job, _ := findJobList(cl)
+	err := cl.Update(context.TODO(), &job)
+	if err != nil {
+		log.Error(err, "Update job", "Failed to Updating the job status")
+	}
+	r.Reconcile(req)
+	jobCount, err := findJobList(cl)
 
-	if job == numberOfJobs {
+	if err != nil {
+		log.Error(err, "Job list", "Failed to fetch job list")
+	}
+	if jobCount == numberOfJobs {
```

Comment:
I believe it would be better to remove `numberOfJobs` variable, and just check here `if jobCount > 1`. If I understand correctly, in this test we want to verify that no new jobs were created, regardless of exact number - too many jobs (more than 2) would be also incorrect.

Reply:
changes done